datalad/support/s3.py

```diff
@@ -291,8 +293,44 @@ def gen_bucket_test2_obscurenames_versioned():
     files("f &$=@:+,?;")
 
 
+def unparse_versioned_url(parsed, version, replace=False):
```

Comment:
Oh... I guess that all code we had was before we came up with our RI helpers... Just instantiate URL(urlstring), and it should take care about those options, so you should be able to change it and then just concert back to string. Sorry that I forgot to point you to the URL class

Reply:
Are you asking that I put the url back together with `URL` and that I replace `get_versioned_url`s use of `urlparse` with `URI`, or just the former?

Reply:
The latter- that we could use URL instead of "manual" urlparse